spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/TextToLabeledSentence.scala

```diff
@@ -60,6 +60,30 @@ class TextToLabeledSentence[T: ClassTag](dictionary: Dictionary)
   }
 }
 
+object TextToLabeledSentence2 {
+  def apply[T: ClassTag](dictionary: Dictionary)
+                        (implicit ev: TensorNumeric[T])
+  : TextToLabeledSentence2[T] =
+    new TextToLabeledSentence2[T](dictionary)
+}
+
+class TextToLabeledSentence2[T: ClassTag](dictionary: Dictionary)
+                                        (implicit ev: TensorNumeric[T])
+  extends Transformer[Array[String], LabeledSentence[T]] {
+  private val buffer = new LabeledSentence[T]()
+
+  override def apply(prev: Iterator[Array[String]]): Iterator[LabeledSentence[T]] = {
+    prev.map(sentence => {
+      val indexes = sentence.map(x =>
+        ev.fromType[Int](dictionary.getIndex(x)))
+      val nWords = indexes.length / 2.0
```

Comment:
If nWords is an odd integer, will drop the middle word in sentence ?
Is it correct?